aten/src/ATen/native/native_functions.yaml

```diff
@@ -1210,7 +1210,7 @@
   supports_named_tensor: True
   variants: function, method
 
-- func: expm1.dtype(Tensor self, ScalarType dtype) -> Tensor
+- func: expm1.dtype(Tensor self, *, ScalarType dtype) -> Tensor
```

Comment:
@nairbv - This enables keyword-only argument feature for `expm1` and tests OK. Would like to hear your comments on this.

1. This enables `dtype` handling, but as I mentioned [before](https://github.com/pytorch/pytorch/pull/33063#issuecomment-583661087) - this does _not_ change any promotion logic for the ops, except allowing the dtype promotion for supported ops with supported dtypes.

For example:
```
x = torch.tensor(2.3, dtype=torch.float16, device="cuda")
output = torch.expm1(x, dtype=torch.float32)
print(output, output.dtype)
```
Outputs: `tensor(8.9820, device="cuda:0"), torch.float32` (expected)

While:

- `torch.expm1(x, torch.float32) # not kwonly friendly` gives an error (expected):

```
TypeError: expm1() received an invalid combination of arguments - got (Tensor, torch.dtype), but expected one of:
 * (Tensor input, Tensor out)
      didn't match because some of the arguments have invalid types: (Tensor, torch.dtype)
 * (Tensor input, torch.dtype dtype)
      didn't match because some of the arguments have invalid types: (Tensor, torch.dtype)
```
- Trying to promote `fp16` to `int` dtype also throws an error (expected): `RuntimeError: result type Half can't be cast to the desired output type Int`.

2. The `out=...` handling for dtype promotion is absent for now. This can be one of the TODOs, depending on your comments & user requirements. 

For example, doing: `torch.expm1(x, out=torch.zeros((), dtype=torch.float32))` throws a `RuntimeError`: `RuntimeError: expected dtype Float but got dtype Half`. This is because the function called when `out=...` is passed does not go through our call chain with `dtype` argument.

Will appreciate your review & comments on this!

Reply:
> This enables keyword-only argument feature for expm1 and tests OK. Would like to hear your comments on this.

Yes I think the `, *, ` syntax is the correct function definition, though I think we'll want `ScalarType? dtype=None` so that it is optional. This matches `torch.mean` and `prod`, as well as `numpy.expm1`.

> This enables dtype handling, but as I mentioned before - this does not change any promotion logic for the ops, except allowing the dtype promotion for supported ops with supported dtypes.

I think we'll expect the behavior to conform to how these options are used in other functions and/or in numpy.

In the docs for ops like torch.prod/sum/mean:
`dtype (torch.dtype, optional) – the desired data type of returned tensor. If specified, the input tensor is casted to dtype before the operation is performed. This is useful for preventing data type overflows. Default: None.`

> TypeError: expm1() received an invalid combination of arguments - got (Tensor, torch.dtype)

This looks correct. dtype as a positional arg shouldn't be supported here.

> doing: torch.expm1(x, out=torch.zeros((), dtype=torch.float32)) throws a RuntimeError

I think we'll need to support this. We'll have to ensure we've set appropriate flags to make the correct checks in validate_dtype etc in TensorIterator.

Reply:
@nairbv - Thanks for the comments. Binary Ops use `PROMOTE` flag by default and is there any reason why Unary Ops should not use the `PROMOTE` flag by default? As far as we can tell, the current solution works fine for tensors on both CPU and CUDA devices.

Reply:
Binary ops iterate through a list containing their two operands to determine a common output type (using type promotion rules implemented in `at::result_type`).

You're modifying a unary op (only one operand, no list), so it seems unclear what behavior you're trying to control through use of this flag.